File: 0066-plus-one/0066-plus-one.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* plusOne(int* digits, int digitsSize, int* returnSize){
int *a = digits;
int n = digitsSize;
int *r = (int*)calloc((n+1),sizeof(int));
int i=n-1;
int j=n;
while(i>=0)
{
    int x = a[i] + 1;
    if(x < 10)
    {
        a[i] = x;
        r = a;
        *returnSize = n;
        return r;
    }
    r[j--]=0;
    a[i]=0;
    if(j==0)
    r[j]=1;
    i--;
}
*returnSize = n+1;
return r;
}

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        List<Integer> list = new ArrayList<>();
        int k = 0;int l = 1;
        for(int i=n-1;i>=0;i--)
        {
            k = digits[i] + k / 10;
            if(l == 1){
            k++;
            l++;}
            list.add(k % 10);
        }
        if(k / 10 != 0)
        list.add(k/10);
        int s = list.size();
        int res[] = new int[s];
        int j = 0;
        for(int i = s-1;i>=0;i--)
        res[j++] = list.get(i);
        return res;
        }
}